cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/monitor/GceRemoteAuthenticator.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.monitor;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import io.cdap.cdap.common.internal.remote.RemoteAuthenticator;
+import io.cdap.common.http.HttpRequest;
+import io.cdap.common.http.HttpRequests;
+import io.cdap.common.http.HttpResponse;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link RemoteAuthenticator} that authenticate remote calls using Google Cloud token acquired from GCE metadata.
+ */
+@SuppressWarnings("unused")
+public class GceRemoteAuthenticator extends RemoteAuthenticator {
```

Comment:
could you please file a jira to move it to a separate extension and add it as todo here?

Reply:
Done. CDAP-16583